Source/Core/DolphinQt2/MenuBar.cpp

```diff
@@ -14,19 +15,152 @@ MenuBar::MenuBar(QWidget* parent)
 	addMenu(tr("Tools"));
 	AddViewMenu();
 	AddHelpMenu();
+
+	EmulationStopped();
+}
+
+void MenuBar::EmulationStarted()
+{
+	// Emulation
+	m_play_action->setEnabled(false);
+	m_play_action->setVisible(false);
+	m_pause_action->setEnabled(true);
+	m_pause_action->setVisible(true);
+	m_stop_action->setEnabled(true);
+	m_reset_action->setEnabled(true);
+	m_fullscreen_action->setEnabled(true);
+	m_frame_advance_action->setEnabled(true);
+	m_screenshot_action->setEnabled(true);
+	m_state_load_menu->setEnabled(true);
+	m_state_save_menu->setEnabled(true);
+	UpdateStateSlotMenu();
+}
+void MenuBar::EmulationPaused()
+{
+	m_play_action->setEnabled(true);
+	m_play_action->setVisible(true);
+	m_pause_action->setEnabled(false);
+	m_pause_action->setVisible(false);
+}
+void MenuBar::EmulationStopped()
+{
+	// Emulation
+	m_play_action->setEnabled(true);
+	m_play_action->setVisible(true);
+	m_pause_action->setEnabled(false);
+	m_pause_action->setVisible(false);
+	m_stop_action->setEnabled(false);
+	m_reset_action->setEnabled(false);
+	m_fullscreen_action->setEnabled(false);
+	m_frame_advance_action->setEnabled(false);
+	m_screenshot_action->setEnabled(false);
+	m_state_load_menu->setEnabled(false);
+	m_state_save_menu->setEnabled(false);
+	UpdateStateSlotMenu();
 }
 
 void MenuBar::AddFileMenu()
 {
 	QMenu* file_menu = addMenu(tr("File"));
-	file_menu->addAction(tr("Open"), this, SIGNAL(Open()));
-	file_menu->addAction(tr("Exit"), this, SIGNAL(Exit()));
+	m_open_action = file_menu->addAction(tr("Open"), this, SIGNAL(Open()));
+	m_exit_action = file_menu->addAction(tr("Exit"), this, SIGNAL(Exit()));
+}
+
+void MenuBar::AddEmulationMenu()
+{
+	QMenu* emu_menu = addMenu(tr("Emulation"));
+	m_play_action = emu_menu->addAction(tr("Play"), this, SIGNAL(Play()));
+	m_pause_action = emu_menu->addAction(tr("Pause"), this, SIGNAL(Pause()));
+	m_stop_action = emu_menu->addAction(tr("Stop"), this, SIGNAL(Stop()));
+	m_reset_action = emu_menu->addAction(tr("Reset"), this, SIGNAL(Reset()));
+	m_fullscreen_action = emu_menu->addAction(tr("Fullscreen"), this, SIGNAL(Fullscreen()));
+	m_frame_advance_action = emu_menu->addAction(tr("Frame Advance"), this, SIGNAL(FrameAdvance()));
+	m_screenshot_action = emu_menu->addAction(tr("Take Screenshot"), this, SIGNAL(Screenshot()));
+	AddStateLoadMenu(emu_menu);
+	AddStateSaveMenu(emu_menu);
+	AddStateSlotMenu(emu_menu);
+	UpdateStateSlotMenu();
+}
+
+void MenuBar::AddStateLoadMenu(QMenu* emu_menu)
+{
+	m_state_load_menu = emu_menu->addMenu(tr("Load State"));
+	m_state_load_menu->addAction(tr("Load State from File"), this, SIGNAL(StateLoad()));
+	m_state_load_menu->addAction(tr("Load State from Selected Slot"), this, SIGNAL(StateLoadSlot()));
+	m_state_load_slots_menu = m_state_load_menu->addMenu(tr("Load State from Slot"));
+	m_state_load_menu->addAction(tr("Undo Load State"), this, SIGNAL(StateLoadUndo()));
+
+	QAction* action;
+	for (int i = 1; i <= 10; i++)
+	{
+		action = m_state_load_slots_menu->addAction(QStringLiteral(""));
```

Comment:
I missed these in my previous review. The variable declaration should be joined at the assignments. There's no need to expose it in an outer scope. Same for all the similar cases below. LGTM for real this time one those are addressed.